plugins/module_utils/openssh/backends/common.py

```diff
@@ -75,7 +76,13 @@ def __init__(self, module):
         self.check_mode = self.module.check_mode
 
     def execute(self):
-        self._execute()
+        try:
+            self._execute()
+        except Exception as e:
+            self.module.fail_json(
+                msg="module encountered exception: %s" % to_native(e),
```

Comment:
I don't think this will help for #416, since the module utils code directly calls `module.fail_json()`. There is no exception that can be caught.

Reply:
See https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/basic.py#L1004

Reply:
Yes, I noticed that after the fact, but there are other places where `OSError` or `IOError` are raised from the `AnsibleModule` methods that could be caught (I should really go through and re-raise them as `OpenSSHError`, but this works for now)

Reply:
Why not provide the traceback (`exception=traceback.format_exc()`)? That makes it a lot easier to identify *where* something went wrong.

Reply:
`fail_json` already does this if the user requests it:
https://github.com/ansible/ansible/blob/0c4c18bc04c562755a368df67fce943ca15418ee/lib/ansible/module_utils/basic.py#L1521-L1529

I decided not to preempt that behavior for UX consistency.

Reply:
Hmm, but this is for unexpected exceptions (that should not reach this point), not for regular errors? So always adding the traceback (and not only with extra verbosity) is probably a good idea.

Reply:
Done